src/room/vault-builder.cpp

```diff
@@ -49,7 +49,9 @@ void vault_monsters(PlayerType *player_ptr, const Pos2D &pos_center, int num)
             }
 
             floor.monster_level = floor.base_level + 2;
-            (void)place_random_monster(player_ptr, pos.y, pos.x, PM_ALLOW_SLEEP | PM_ALLOW_GROUP);
+            if (place_random_monster(player_ptr, pos.y, pos.x, PM_ALLOW_SLEEP | PM_ALLOW_GROUP)) {
+                break;
+            }
             floor.monster_level = floor.base_level;
```

Comment:
`floor.monster_level` is a data member that specifies the monster level to be created. The reason why `floor.base_level + 2` is set immediately before is to boost the creation level for the vault. Therefore, after calling `place_random_monster()`, `floor.monster_level` must be restored, and this code would cause the function to exit with the monster_level boosted, which could affect the monster creation processing later.

As shown below, it is better to receive the result in a variable and restore floor.monster_level before making the decision to exit the loop.

```c++
floor.monster_level = floor.base_level + 2;
const auto has_placed = place_random_monster(…);
floor.monster_level = floor.base_level;
if (has_placed) {
    break;
}
```